src/xenia/kernel/xboxkrnl/xboxkrnl_hal.cc

```diff
@@ -18,16 +18,56 @@ namespace kernel {
 namespace xboxkrnl {
 
 void HalReturnToFirmware_entry(dword_t routine) {
-  // void
-  // IN FIRMWARE_REENTRY  Routine
-
-  // Routine must be 1 'HalRebootRoutine'
-  assert_true(routine == 1);
-
   // TODO(benvank): diediedie much more gracefully
   // Not sure how to blast back up the stack in LLVM without exceptions, though.
-  XELOGE("Game requested shutdown via HalReturnToFirmware");
-  exit(0);
+  switch (routine) {
```

Comment:
Instead of making this whole switch case for each entry you can make an constexpr array with all these strings